adage-front/src/app/components/OffersSearch/Offers/Pagination/pagination_calculator.ts

```diff
@@ -0,0 +1,29 @@
+export const getPages = (
+  currentRefinement: number,
+  nbPages: number
+): Array<string | number> => {
+  let pages: (number | string)[] = [1]
+  const pagesToDisplay = [
+    currentRefinement - 1,
+    currentRefinement,
+    currentRefinement + 1,
+  ].filter(page => page > 1 && page < nbPages)
+
+  if (pagesToDisplay[0] > 2) {
+    pages.push('...')
+    pages.push(...pagesToDisplay)
+    if (pagesToDisplay[pagesToDisplay.length - 1] < nbPages - 1) {
+      pages.push('...')
+    }
+    pages.push(nbPages)
+  } else if (nbPages > 3) {
+    pages.push(...pagesToDisplay)
+    if (pagesToDisplay[pagesToDisplay.length - 1] < nbPages - 1) {
+      pages.push('...')
+    }
+    pages.push(nbPages)
+  } else {
+    pages = new Array(nbPages).fill(null).map((_, index) => index + 1)
+  }
+  return pages
+}
```

Comment:
Je comprend pas grand chose à ce code, peut-être un mélange de nomage hasardeux et de complexité abstraite... y'a peut-être moyen de rendre ça plus intelligible ? on peut prendre 5 minutes si c'est pas trop préssé, sinon on peut vivre avec...

Reply:
Je vais voir pour refacto

Reply:
pas sur que se soit plus clair : 
```suggestion
export const getPages = (
  currentPage: number,
  totalPages: number
): Array<'...' | number> => {
  const firstPage = 1;
  const lastPage = totalPages;
  
 const pagesBeforeCurrent = new Set(Array(currentPage - 1).fill('...').map((dots, index) => {
   const page = index + 1;
   return (page === firstPage || page === currentPage - 1 || page === currentPage) 
     ? page 
     : dots
 }))
 
 const pagesAfterCurrent = new Set(Array(totalPages - currentPage + 1).fill('...').map((dots, index) => {
   const page = index + 1;
   return (page === lastPage || page === currentPage + 1) 
     ? page 
     : dots
 }))
 
 return [
   ...Array.from(pagesBeforeCurrent),
   currentPage,
   ...Array.from(pagesAfterCurrent)
 ]
}
```